File: 卡码/代码随想录/1/06-二叉树/Python/02-二叉树属性/LeetCode_0101.py
#101. 对称二叉树
from typing import *
from ..TreeNode import *

class Solution:
  def compare(self,left:Optional[TreeNode],right:Optional[TreeNode])->bool:
    if left!=None and right==None:
      return False
    elif left==None and right!=None:
      return False
    elif left==None and right==None:
      return True
    elif left.val!=right.val:
      return False
    else:
      return self.compare(left.left,right.right) and self.compare(left.right,right.left)

  def isSymmetric(self, root: Optional[TreeNode]) -> bool:
    if not root:
      return False
    return self.compare(root.left,root.right)

File: 卡码/代码随想录/1/06-二叉树/Js/02-二叉树属性/LeetCode_0101.js
//101. 对称二叉树
import TreeNode from "../TreeNode.js";

const compare=(left,right)=>{
  if(left&&!right) return false;
  else if(!left&&right) return false;
  else if(!left&&!right) return true;
  else if(left.val!=right.val) return false;
  else return compare(left.left,right.right)&&compare(left.right,right.left);
};

const isSymmetric = function(root) {
  if(!root) return false;
  return compare(root.left,root.right);
};